src/platform/linux/wayland/keyboard.rs

```diff
@@ -120,45 +127,47 @@ pub fn init_keyboard(
-                }
-                wl_keyboard::Event::Key { key, state, .. } => {
-                    if let Some(wid) = target {
-                        let state = match state {
-                            wl_keyboard::KeyState::Pressed => ElementState::Pressed,
-                            wl_keyboard::KeyState::Released => ElementState::Released,
-                        };
-                        my_sink.lock().unwrap().send_event(
-                            WindowEvent::KeyboardInput {
-                                device_id: ::DeviceId(::platform::DeviceId::Wayland(DeviceId)),
-                                input: KeyboardInput {
-                                    state: state,
-                                    scancode: key,
-                                    virtual_keycode: None,
-                                    modifiers: ModifiersState::default(),
+            seat.get_keyboard(|keyboard| {
+                keyboard.implement(move |evt, _| match evt {
+                    wl_keyboard::Event::Enter { surface, .. } => {
+                        let wid = make_wid(&surface);
+                        my_sink
+                            .lock()
+                            .unwrap()
+                            .send_event(WindowEvent::Focused(true), wid);
+                        target = Some(wid);
+                    }
+                    wl_keyboard::Event::Leave { surface, .. } => {
+                        let wid = make_wid(&surface);
+                        my_sink
+                            .lock()
+                            .unwrap()
+                            .send_event(WindowEvent::Focused(false), wid);
+                        target = None;
+                    }
+                    wl_keyboard::Event::Key { key, state, .. } => {
+                        if let Some(wid) = target {
+                            let state = match state {
+                                wl_keyboard::KeyState::Pressed => ElementState::Pressed,
+                                wl_keyboard::KeyState::Released => ElementState::Released,
+                            };
+                            my_sink.lock().unwrap().send_event(
+                                WindowEvent::KeyboardInput {
+                                    device_id: ::DeviceId(::platform::DeviceId::Wayland(DeviceId)),
+                                    input: KeyboardInput {
+                                        state: state,
+                                        scancode: key,
+                                        virtual_keycode: None,
+                                        modifiers: ModifiersState::default(),
+                                    },
                                 },
-                            },
-                            wid,
-                        );
+                                wid,
+                            );
+                        }
                     }
```

Comment:
We probably  should handle `wl_keyboard::Event::Modifiers` too, to keep symmetry with the SCTK impl.

Reply:
Actually, no need, I don't think it's possible that we receive this event if `libxkbcommon` is not available, given it uses its format.